libs/sdk-core/src/breez_services.rs

```diff
@@ -230,16 +231,24 @@ impl BreezServices {
     pub async fn send_payment(&self, bolt11: String, amount_sats: Option<u64>) -> Result<Payment> {
         self.start_node().await?;
         let parsed_invoice = parse_invoice(bolt11.as_str())?;
-        let payment_res = self
-            .node_api
-            .send_payment(bolt11.clone(), amount_sats)
-            .await;
-        self.on_payment_completed(
-            parsed_invoice.payee_pubkey.clone(),
-            Some(parsed_invoice),
-            payment_res,
-        )
-        .await
+        match self
+            .persister
+            .get_completed_payment_by_hash(&parsed_invoice.payment_hash)?
+        {
+            Some(_) => Err(anyhow!("Invoice already paid")),
```

Comment:
Perhaps we can introduce a new SdkError here?

Reply:
Done. Added SdkError and updated test